src/main/java/net/fabricmc/loom/task/MigrateMappingsTask.java

```diff
@@ -149,20 +148,21 @@ private File loadMappings() {
 		return Iterables.getOnlyElement(files);
 	}
 
-	private static TinyTree getMappings(File mappings) throws IOException {
+	private static MemoryMappingTree getMappings(File mappings) throws IOException {
 		Path temp = Files.createTempFile("mappings", ".tiny");
 
 		try (FileSystem fileSystem = FileSystems.newFileSystem(mappings.toPath(), (ClassLoader) null)) {
 			Files.copy(fileSystem.getPath("mappings/mappings.tiny"), temp, StandardCopyOption.REPLACE_EXISTING);
 		}
 
-		try (BufferedReader reader = Files.newBufferedReader(temp)) {
-			return TinyMappingFactory.loadWithDetection(reader);
-		}
+		MemoryMappingTree mappingTree = new MemoryMappingTree();
+		MappingReader.read(temp, mappingTree);
+
+		return mappingTree;
```

Comment:
(A bit off-topic for this PR) Couldn't this read the mappings from the jar directly since that uses `Path`s too?

Reply:
Done